src/chordsymbol.ts

```diff
@@ -395,227 +222,123 @@ export class ChordSymbol extends Modifier {
     return true;
   }
 
-  /** Currently unused. */
-  protected static noFormat: boolean = false;
-
   protected symbolBlocks: ChordSymbolBlock[] = [];
   protected horizontal: number = ChordSymbolHorizontalJustify.LEFT;
   protected vertical: number = ChordSymbolVerticalJustify.TOP;
-  protected useKerning: boolean = true;
-  protected reportWidth: boolean = true;
-
-  // Initialized by the constructor via this.setFont().
-  protected textFormatter!: TextFormatter;
 
   constructor() {
     super();
-    if (Tables.currentMusicFont().getName() === 'Petaluma') {
-      // Fixes Issue #1180
-      this.textFont.family = 'PetalumaScript, Arial, sans-serif';
-    }
-
-    this.textFormatter = TextFormatter.create(this.textFont);
   }
 
   /**
    * The offset is specified in `em`. Scale this value by the font size in pixels.
    */
   get superscriptOffset(): number {
-    return ChordSymbol.superscriptOffset * this.textFormatter.fontSizeInPixels;
+    return ChordSymbol.superscriptOffset * Font.convertSizeToPixelValue(this.textFont?.size);
   }
 
   get subscriptOffset(): number {
-    return ChordSymbol.subscriptOffset * this.textFormatter.fontSizeInPixels;
-  }
-
-  setReportWidth(value: boolean): this {
-    this.reportWidth = value;
-    return this;
-  }
-
-  getReportWidth(): boolean {
-    return this.reportWidth;
-  }
-
-  updateOverBarAdjustments(): void {
-    const barIndex = this.symbolBlocks.findIndex(
-      ({ symbolType, glyph }: ChordSymbolBlock) =>
-        symbolType === SymbolTypes.GLYPH && glyph !== undefined && glyph.code === 'csymDiagonalArrangementSlash'
-    );
-
-    if (barIndex < 0) {
-      return;
-    }
-    const bar = this.symbolBlocks[barIndex];
-    const xoff = bar.width / 4;
-    const yoff = 0.25 * this.textFormatter.fontSizeInPixels;
-    let symIndex = 0;
-    for (symIndex === 0; symIndex < barIndex; ++symIndex) {
-      const symbol = this.symbolBlocks[symIndex];
-      symbol.xShift = symbol.xShift + xoff;
-      symbol.yShift = symbol.yShift - yoff;
-    }
-
-    for (symIndex = barIndex + 1; symIndex < this.symbolBlocks.length; ++symIndex) {
-      const symbol = this.symbolBlocks[symIndex];
-      symbol.xShift = symbol.xShift - xoff;
-      symbol.yShift = symbol.yShift + yoff;
-    }
-  }
-
-  updateKerningAdjustments(): void {
```

Comment:
I'm OK with removing the kerning code for now. If we ever decide we need it in V5, we can come back to this PR and see if the V4 approach can help us.

Reply:
ok